linode/manager
# upcoming: [DI-20357] - Changes for ACLP Dashboard with Filters component

## Description 📝
Added a new CloudPulseDashboardWithFilters component which has the capability to build the filters needed in service provider page as well as the dashboard. This component will be integrated in the service provider pages.

## Changes  🔄

1. Added new CloudPulseDashboardWithFilters component that has the capability to build the filters needed in service provider page as well as the dashboard. This will be integrated only in the service provider pages.
2. Modified the dbaas config to introduce a new filter nodeType and changed needInServicePage prop to false

## Target release date 🗓️
03-09-2024

## Preview 📷
| Before  | After   |
| ------- | ------- |
|![Screenshot 2024-08-28 at 11 40 04 AM](https://github.com/user-attachments/assets/ff356cfc-8dd6-44ad-bb7e-f565a49e56b5)|![Screenshot 2024-08-28 at 11 40 29 AM](https://github.com/user-attachments/assets/01fed9ef-ddad-46f2-bdb2-7b7df044659f)|
|![Screenshot 2024-08-28 at 11 40 38 AM](https://github.com/user-attachments/assets/ecd7582f-d271-472e-a2b0-1d991bd16cd0)||
|![Screenshot 2024-08-28 at 11 41 19 AM](https://github.com/user-attachments/assets/d8a1c9ca-077c-4c66-9556-0da4a5248f37)|![Screenshot 2024-08-28 at 11 41 12 AM](https://github.com/user-attachments/assets/c4672c4b-443f-498d-9b90-49c9adcee4ad)|

## How to test 🧪
To test this, we need to integrate the CloudPulseDashboardWithFilters in a service Provider page. The below commit shows the test integration we made in database details index page 

https://github.com/venkymano/manager/commit/e0f3598630fa4c66e493d76b07dea515c1334f0c

1. Copy the code in the commit in the same file mentioned in the commit
2. Start the cloud manager UI application
3. Login and enable mock user as some API endpoints are not available still.
4. Navigate to any database instance details page, and navigate to the monitor tab.
5. You will see filters needed in service provide page as well as a mock dashboard rendered.

**Note** - We support only linode and dbaas, so the sample integration revolves around Managed Databases


## As an Author I have considered 🤔

*Check all that apply*
